apps/src/netsim/NetSimPacketSizeControl.js

```diff
@@ -0,0 +1,108 @@
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxstatements: 200
+ */
+/* global $ */
+'use strict';
+
+var markup = require('./NetSimPacketSizeControl.html');
+
+/**
+ * @type {number}
+ * @const
+ */
+var SLIDER_INFINITY_VALUE = 1025;
+
+/**
+ * Generator and controller for packet size slider/selector
+ * @param {jQuery} rootDiv
+ * @param {function} packetSizeChangeCallback
+ * @constructor
+ */
+var NetSimPacketSizeControl = module.exports = function (rootDiv,
+    packetSizeChangeCallback) {
+  /**
+   * Component root, which we fill whenever we call render()
+   * @type {jQuery}
+   * @private
+   */
+  this.rootDiv_ = rootDiv;
+
+  /**
+   * @type {function}
+   * @private
+   */
+  this.packetSizeChangeCallback_ = packetSizeChangeCallback;
+
+  /**
+   * Internal state
+   * @type {number}
+   * @private
+   */
+  this.currentPacketSize_ = Infinity;
+
+  this.render();
+};
+
+/**
+ * Fill the root div with new elements reflecting the current state
+ */
+NetSimPacketSizeControl.prototype.render = function () {
+  var renderedMarkup = $(markup({}));
+  this.rootDiv_.html(renderedMarkup);
+  this.rootDiv_.find('.packet_size_slider').slider({
```

Comment:
Hah, read through all of the code and was like, wait, where'd the slider happen? Then saw this line! Nice reuse.